packages/loader/container-loader/src/container.ts

```diff
@@ -815,17 +817,16 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                 this._attachState = AttachState.Attaching;
                 this.emit("attaching");
             }
-            assert(!!this.cachedAttachSummary,
+            assert(!!this.cachedAttachSummary || this.loader.services.detachedBlobStorage?.size !== 0,
                 0x0d7 /* "Summary should be there either by this attach call or previous attach call!!" */);
 
             const createNewResolvedUrl = await this.urlResolver.resolve(request);
             ensureFluidResolvedUrl(createNewResolvedUrl);
-            const summary = this.cachedAttachSummary;
             // Actually go and create the resolved document
             if (this.service === undefined) {
                 this.service = await runWithRetry(
                     async () => this.serviceFactory.createContainer(
-                        summary,
+                        this.cachedAttachSummary as unknown as ISummaryTree,
```

Comment:
Is the purpose of this cast to get rid of undefined?
If yes, I'd rather assert it's not undefined before it - that way compiler is happy, and if we got it wrong, we would get good telemetry data (as opposed to cryptic failures down the road)

Reply:
Oh, undefined is expected, you you are missing updating IDocumentServiceFactory.createContainer to accept undefined here.